cmd/data-node/commands/networkhistory/latest_history_segment.go

```diff
@@ -56,47 +45,36 @@ func (cmd *latestHistorySegment) Execute(_ []string) error {
 		os.Exit(1)
 	}
 
-	if !datanodeLive(cmd.Config) {
-		return fmt.Errorf("datanode must be running for this command to work")
-	}
+	ctx, cancelFn := context.WithCancel(context.Background())
+	defer cancelFn()
 
-	client, conn, err := getDatanodeClient(cmd.Config)
+	networkHistoryStore, err := store.New(ctx, log, cmd.Config.ChainID, cmd.Config.NetworkHistory.Store, vegaPaths.StatePathFor(paths.DataNodeNetworkHistoryHome), false)
 	if err != nil {
-		handleErr(log, cmd.Output.IsJSON(), "failed to get datanode client", err)
+		handleErr(log, cmd.Output.IsJSON(), "failed to create network history store", err)
 		os.Exit(1)
 	}
-	defer func() { _ = conn.Close() }()
 
-	resp, err := client.GetActiveNetworkHistoryPeerAddresses(context.Background(), &v2.GetActiveNetworkHistoryPeerAddressesRequest{})
+	segments, err := networkHistoryStore.ListAllIndexEntriesOldestFirst()
```

Comment:
@ettec Hey, I think there is a slight problem with using the store directly when the node is running. It seems like the file is being used by node and it's locked somehow.